bse/driver-jack.cc

```diff
@@ -0,0 +1,905 @@
+	{
+	  const char *end = strchr (jack_ports[i], ':');
+	  if (end)
+	    {
+              std::string device_name (jack_ports[i], end);
+
+	      jack_port_t *jack_port = jack_port_by_name (jack_client, jack_ports[i]);
+	      if (jack_port)
+		{
+                  const char *port_type = jack_port_type (jack_port);
+                  if (strcmp (port_type, JACK_DEFAULT_AUDIO_TYPE) == 0)
+                    {
+	              DeviceDetails &details = devices[device_name];
+	              details.ports++;
+
+                      int flags = jack_port_flags (jack_port);
+                      if (flags & JackPortIsInput)
+                        {
+                          details.input_ports++;
+                          details.input_port_names.push_back (jack_ports[i]);
+                        }
+                      if (flags & JackPortIsOutput)
+                        {
+                          details.output_ports++;
+                          details.output_port_names.push_back (jack_ports[i]);
+                        }
+                      if (flags & JackPortIsPhysical)
+                        {
+                          details.physical_ports++;
+
+                          if (!have_default_device)
+                            {
+                              /* the first device that has physical ports is the default device */
+                              details.default_device = true;
+                              have_default_device = true;
+                            }
+                        }
+                      if (flags & JackPortIsTerminal)
+                        details.terminal_ports++;
+                    }
+		}
+	    }
+	}
+      free (jack_ports);
+    }
+
+  return devices;
+}
+
+static void
+list_jack_drivers (Driver::EntryVec &entries)
+{
+  map<std::string, DeviceDetails> devices;
+  jack_client_t *jack_client = connect_jack();
+  if (jack_client)
+    {
+      devices = query_jack_devices (jack_client);
+      disconnect_jack (jack_client);
+    }
+  else
+    {
+      // should we try to generate and show an error message if connecting jack failed?
+    }
+
+  for (std::map<std::string, DeviceDetails>::iterator di = devices.begin(); di != devices.end(); di++)
+    {
+      const std::string &device_name = di->first;
+      DeviceDetails &details = di->second;
+
+      /* the default device is usually the hardware device, so things should work as expected
+       * we could show try to show non-default devices as well, but this could be confusing
+       */
+      if (details.default_device)
+        {
+          Driver::Entry entry;
+          entry.devid = device_name;
+          entry.priority = Driver::JACK;
+          entries.push_back (entry);
+        }
+    }
+}
+
+/* macro for jack dropout tests - see below */
+#define TEST_DROPOUT() if (unlink ("/tmp/drop") == 0) usleep (1.5 * 1000000. * buffer_frames_ / mix_freq_); /* sleep 1.5 * buffer size */
+
```

Comment:
I know this is disabled, atm, but please still make this /tmp/bse-dropout (or similar) so we're not stamping over namespaces if this is activated on purpose or by accident.

Reply:
Ok, done.